content/recipes/authoring/kata-snippets/description.md

```diff
@@ -5,6 +5,54 @@ prev: /recipes/authoring/kata-snippets/
+
+
+## Structure
+
+- **Description should be structured into meaningfull parts**, for example:
+  - overwiew, general context, or introduction,
+  - general statement of a task user has to solve,
+  - description of input and output, potentially with examples,
+  - detailed rules and requirements,
+  - additional notes, explanations, technical details, follow-ups.  
+Above is just an example, not all sections are necessary if not applicable, and authors can use another structuring if they like. However, related pieces of information should be held together: explaining inputs at 3-4 different places isn't a good idea.
+- **Descriptions should be consise, but precise.** Description of a task should be possibly short, because long descriptions suggest that task is either not very well defined, or is very complex. Fabularized or themed descriptions are OK as long as they don't get in a way of understanding the problem.
+- It's usually recommended to describe general aspects first, and then go further into details, edge cases, and finally, exceptions. Descriptions structured this way are usually easier to follow.
+
+
+## Requirements
+
+- **Description should clearly state all requirements** user has to fulfill to complete the attempted kata. Unless the kata is a puzzle, users should not feel surprised by tests which verify some requirements not mentioned in the description. Description should _explicitly_, _very clearly_ state what inputs have to be handled, what inputs do not have to be considered, and how to handle exceptional situations. When a test case fails, there should always be a point in the description which would explain why this particular test failed.
+- Descriptions should provide **examples of inputs and outputs for some representative scenarios**. Provide as many examples as you think necessary.
+  - **Avoid things like _"See sample tests for more examples"_**. To see sample tests, users have to enter the kata trainer, what can mark the kata as unfinished in their profile.
+  - **Use images when helpful.** Currently, Codewars does not allow to attach files to a kata, so image has to be hosted externally, or HTML image with Base-64 encoded content, preferrably SVG, can be used. Make sure that image does not break layout of the kata page.
```

Comment:
```suggestion
  - **Use images when helpful, but prefer text if it's sufficient to communicate the idea.** Currently, Codewars does not allow attaching files to a kata which makes hosting images externally a default solution. Alternatively, a HTML image with Base-64 encoded content or an SVG can be included in the description. This is preferred because external image hosts create a dependency on another website which may become stale over time, rendering the kata unusable. Make sure that image does not break the layout of the kata page. When using images, take into consideration users with small screen widths and visual impairments.
```